pkgs/tools/networking/linux-router/default.nix

```diff
@@ -0,0 +1,102 @@
+# SOURCE: https://github.com/NixOS/nixpkgs/pull/92787/files
+
+{ stdenv, lib, pkgs, fetchFromGitHub
+
+# Dependencies
+, bash , procps , iproute2 , dnsmasq , iptables
+
+# WiFi hotspot dependencies
+, hostapd , iw
+, wirelesstools ? null # for iwconfig
```

Comment:
There is no need to set optional packages `null`. They will not be used if your `optional` in the `binPath` further down is false.

Reply:
I thought that this would make the packages truly optional in that you don't even need them to exist in the repo at all. Is it considered more 'clean' to just require them to exist so the code is less bloated (also regarding the asserts below)

Reply:
The packages are expected to exist in the repo once you turn a use* option. Let's assume one of the optional packages gets removed or renamed: if they are set `null` by default, then the code will not fail until someone actually uses them. Thus it may take a long time until this error gets discovered. My suggestion (which is common practice in nixpkgs), will fail in that case and uncover the error right away.
That the code gets less bloated this way is also nice.

Reply:
Thanks for clarification :smiley_cat: